modules/apps/commerce/commerce-frontend-taglib/src/main/resources/META-INF/resources/price/page.jsp

```diff
@@ -17,125 +17,245 @@
+				}
+
+				if (Validator.isNotNull(discountPercentages[2])) {
+					discountLevel3HideClass = "";
+
+					discountPercentageLevel3 = discountPercentages[2];
+				}
+
+				if (Validator.isNotNull(discountPercentages[3])) {
+					discountLevel4HideClass = "";
+
+					discountPercentageLevel4 = discountPercentages[3];
+				}
+			}
+			%>
+
+			<span class="discount-percentage-level1 price-value-discount <%= discountLevel1HideClass %>" data-text-cp-instance-discount-percentage-level-1>
+				<%= discountPercentageLevel1 %>
 			</span>
-			<span class="<%= showStartingAt ? "" : "hide" %>">
-				<liferay-ui:message key="starting-at" />
+			<span class="discount-percentage-level2 price-value-discount <%= discountLevel2HideClass %>" data-text-cp-instance-discount-percentage-level-2>
+				<%= discountPercentageLevel2 %>
+			</span>
+			<span class="discount-percentage-level3 price-value-discount <%= discountLevel3HideClass %>" data-text-cp-instance-discount-percentage-level-3>
+				<%= discountPercentageLevel3 %>
+			</span>
+			<span class="discount-percentage-level4 price-value-discount <%= discountLevel4HideClass %>" data-text-cp-instance-discount-percentage-level-4>
+				<%= discountPercentageLevel4 %>
 			</span>
 		</span>
-		<span class="product-promo-price <%= Validator.isNull(formattedPromoPrice) ? "" : "price-value-inactive" %>" data-text-cp-instance-price>
-			<%= formattedPrice %>
-		</span>
-		<span class="price-label <%= promoPriceClass %>" data-text-cp-instance-promo-price-label>
-			<%= Validator.isNull(promoPriceLabel) ? LanguageUtil.get(request, "sale-price") : promoPriceLabel %>
-		</span>
-		<span class="product-price <%= promoPriceClass + promoPriceActiveClass %>" data-text-cp-instance-promo-price>
-			<%= formattedPromoPrice %>
+		<span class="discount-percentage price-value price-value-discount <%= discountPercentageHideClass %>" data-text-cp-instance-discount-percentage>
+			-<%= prices.getDiscountPercentage() %>
 		</span>
+	</span>
+	<span class="price-label <%= discountHideClass %>">
+		<liferay-ui:message key="final-price" />
+	</span>
+	<span class="price-value price-value-final <%= discountHideClass %> <%= GetterUtil.getString(additionalDiscountClasses) %>" data-text-cp-instance-final-price>
+		<%= prices.getFinalPrice() %>
+	</span>
+</div>
 
-		<c:if test="<%= commerceDiscountValue != null %>">
+<liferay-portlet:actionURL name="/cp_content_web/check_cp_instance" portletName="com_liferay_commerce_product_content_web_internal_portlet_CPContentPortlet" var="checkCPInstanceURL">
+	<portlet:param name="cpDefinitionId" value="<%= String.valueOf(cpDefinitionId) %>" />
+	<portlet:param name="groupId" value="<%= String.valueOf(themeDisplay.getScopeGroupId()) %>" />
+</liferay-portlet:actionURL>
 
-			<%
-			CommerceMoney discountAmountCommerceMoney = commerceDiscountValue.getDiscountAmount();
-			%>
+<aui:script require="commerce-frontend-js/utilities/eventsDefinitions as events">
```

Comment:
This JS scriptlet can be put in a `<liferay-frontend:component module="...modulePath" />` and encapsulated in a separate JS resource.